File: src/main/scala/com/xmlcalabash/util/SysIdLocation.scala
package com.xmlcalabash.util

import net.sf.saxon.s9api.Location

// Via Florent Georges

class SysIdLocation(sysid: String) extends Location {
  override def getSystemId: String = sysid
  override def getPublicId: String = null
  override def getLineNumber: Int = -1
  override def getColumnNumber: Int = -1
  override def saveLocation(): Location = this
}

File: xmlcalabash/src/main/kotlin/com/xmlcalabash/util/SysIdLocation.kt
package com.xmlcalabash.util

import net.sf.saxon.s9api.Location

class SysIdLocation (val sysId: String): Location {
    override fun getPublicId(): String? = null
    override fun getSystemId(): String? = sysId
    override fun getLineNumber(): Int = -1
    override fun getColumnNumber(): Int = -1
    override fun saveLocation(): Location = this
}